pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -203,9 +208,11 @@ func (rc *reconciler) reconcile() {
 		if rc.actualStateOfWorld.VolumeNodeExists(
 			volumeToAttach.VolumeName, volumeToAttach.NodeName) {
 			// Volume/Node exists, touch it to reset detachRequestedTime
+			// TODO: I don't understand why this is the case.  If we are in the process of detaching something, we "forget" if we desire it to be attached?
```

Comment:
When reconciler issue detach operation, it first set detach request time, so if safeToDetach fails, it remember when it was requested to detach and might timeout for detaching when timeout (We actually still need to examine this logic)

But considering user delete and add pod quickly. So detachRequestTime was set when deleting the pod, but after adding the pod and volume back, this detachRequestTime should be cleared. This is the logic to clear the previous detachRequestTime.